lib/socket.ts

```diff
@@ -758,9 +758,6 @@ export class Socket<
     }
 
     this._cleanup();
```

Comment:
_cleanup is called here and in two more places: https://github.com/socketio/socket.io/blob/main/lib/namespace.ts#L327-L333

I believe the `this.nsp._remove(this);` should be called in all these cases